Code/Editor/Core/EditorActionsHandler.cpp

```diff
@@ -1801,6 +1868,52 @@ void EditorActionsHandler::RefreshLayoutActions()
     }
 }
 
+void EditorActionsHandler::RefreshToolboxMacroActions()
+{
+    // If the toolbox macros are being displayed in the menu already, remove them.
+    m_menuManagerInterface->RemoveActionsFromMenu(ToolBoxMacrosMenuIdentifier, m_toolboxMacroActionIdentifiers);
+    m_toolboxMacroActionIdentifiers.clear();
+
+    // Place all actions in the same sort index in the menu .
+    // This will display them in order of addition (alphabetical).
+    const int sortKey = 0;
+
+    auto tools = GetIEditor()->GetToolBoxManager();
```

Comment:
suggestion: This may be outside the scope of this PR but is there anyway to avoid new calls to `GetIEditor` 😬 It would be wonderful one day to remove that as it's all legacy cruft mostly 🙁 Is there another way (using a Bus or Interface?) to grab the ToolBoxManager if it's sticking around?

Reply:
Honestly, I'm having an extremely hard time holding back from trying to refactor some of the functionality during this process, cause the risk of falling into rabbit holes is too big and the timeline is a bit tight :D 

I am raising some of these questions to UX as appropriate though, especially in terms of whether some of these tools like the quick access bar, the toolbox manager etc. are still necessary, should be revamped etc. The hope is to start removing the reliance on these legacy calls over time.

Reply:
This is tough, but yeah I think I'd lean towards leaving as-is because I'm not sure if the toolbox manager is even going to stick around.

Reply:
> Honestly, I'm having an extremely hard time holding back from trying to refactor some of the functionality during this process, cause the risk of falling into rabbit holes is too big and the timeline is a bit tight :D
> 
> I am raising some of these questions to UX as appropriate though, especially in terms of whether some of these tools like the quick access bar, the toolbox manager etc. are still necessary, should be revamped etc. The hope is to start removing the reliance on these legacy calls over time.

Lol totally understand, and you're 100% doing the right thing, I just didn't want to ignore it and completely understand the reasoning. I think as a separate pass, doing an update to have an interface for ToolboxManager rather than relying on `GetIEditor` would be great but again it definitely does not have to be part of this change. Thanks for the info 👍